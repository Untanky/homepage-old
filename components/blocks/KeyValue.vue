<template>
  <div
    class="text-lg"
    :class="data.classes"
  >
    <span
      class="font-bold"
    >
      {{data.key}}:
    </span>
    <span>
      {{data.value}}
    </span>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import KeyValue from '../../src/components/KeyValue';

export default Vue.extend({
  name: 'KeyValue',
  props: {
    data: {
      type: Object as PropType<KeyValue>,
      required: true,
    },
  },
});
</script>
