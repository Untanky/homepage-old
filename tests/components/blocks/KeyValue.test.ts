/* eslint-disable no-undef */
// global it, expect
import { mount } from '@vue/test-utils';
import KeyValue from '../../../components/blocks/KeyValue.vue';

describe('KeyValue', () => {
  it('should run', () => {
    const props = {
      data: {
        key: 'Key',
        value: 'Value',
      },
    };

    const comp = mount(KeyValue, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });
});
