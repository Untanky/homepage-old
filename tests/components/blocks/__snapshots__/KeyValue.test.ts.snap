// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValue should run 1`] = `
<div
  class="text-lg"
>
  <span
    class="font-bold"
  >
    
    Key:
  
  </span>
   
  <span>
    
    Value
  
  </span>
</div>
`;
